mypy_django_plugin/transformers/models.py

```diff
@@ -151,85 +153,88 @@ def run_with_model_cls(self, model_cls: Type[Model]) -> None:
+        assert isinstance(parent_manager, TypeInfo)
+        tvars = parent_manager.defn.type_vars
 
-                original_base = helpers.reparametrize_instance(original_base, [Instance(self.model_classdef.info, [])])
-            bases.append(original_base)
-
-        new_manager_info = self.add_new_class_for_current_module(name, bases)
-        # copy fields to a new manager
-        new_cls_def_context = ClassDefContext(cls=new_manager_info.defn, reason=self.ctx.reason, api=self.api)
-        custom_manager_type = Instance(new_manager_info, [Instance(self.model_classdef.info, [])])
-
-        for name, sym in base_manager_info.names.items():
-            # replace self type with new class, if copying method
-            if isinstance(sym.node, FuncDef):
-                helpers.copy_method_to_another_class(
-                    new_cls_def_context,
-                    self_type=custom_manager_type,
-                    new_method_name=name,
-                    method_node=sym.node,
-                    original_module_name=base_manager_info.module_name,
-                )
-                continue
+        base = Instance(base_manager_info, tvars)
+
+        new_manager_info = self.add_new_class_for_current_module(name, [base])
+        new_manager_info.defn.type_vars = tvars
+        new_manager_info.add_type_vars()
+
+        return self.model_parametrize(new_manager_info)
+
+    def _is_declared_inside_class(self, manager_name: str) -> bool:
+        """
+        Ignore managers from wrong scope.
+
+        Check what do we have currently. If it is `AnyType`, then
+        it has to be generated by our plugin
+        (see `create_new_manager_class_from_from_queryset_method`
+        and `fail_if_manager_type_created_in_model_body` hooks)
+        """
+        type_now = self.model_classdef.info.names[manager_name].type
+        return isinstance(type_now, AnyType) and type_now.type_of_any == TypeOfAny.implementation_artifact
+
+    def _should_subclass_manager(self, manager: TypeInfo) -> bool:
+        """
+        Check if new manager instance should be created.
 
-            new_sym = sym.copy()
-            if isinstance(new_sym.node, Var):
-                new_var = Var(name, type=sym.type)
-                new_var.info = new_manager_info
-                new_var._fullname = new_manager_info.fullname + "." + name
-                new_sym.node = new_var
-            new_manager_info.names[name] = new_sym
+        Ending up here could for instance be due to having a custom _Manager_
+        that is not built from a custom QuerySet. Another example is a
+        related manager.
+        Don't interfere with dynamically generated manager classes
+        """
 
-        return custom_manager_type
+        is_dynamic = manager.metadata.get("django", {}).get("from_queryset_manager")
+        return self.has_any_parametrized_manager_as_base(manager) and not is_dynamic
```

Comment:
```suggestion
        return self.has_manager_with_implicit_any_parameter_as_base(manager) and not is_dynamic
```

Reply:
Maybe just inline this? It is used in single place, and function name doesn't read better than `any(map(...))` directly.